File: src/gui/position.rs
#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
pub struct Position(pub u32, pub u32);

impl Position
{
    pub fn raw(&self) -> (u32, u32)
    {
        (self.0, self.1)
    }
}

impl From<(u32, u32)> for Position {
    fn from((x, y): (u32, u32)) -> Position {
        Position(x, y)
    }
}

File: src/gui/position.zig
pub const Position = @This();

x: u32,
y: u32,

pub fn new(x: u32, y: u32) Position
{
    return Position { .x = x, .y = y };
}

pub fn raw(self: Position) .{ u32, u32 } {
    return .{ self.x, self.y };
}
